app/src/main/java/com/babylon/wallet/android/presentation/transaction/analysis/processor/ValidatorStakeProcessor.kt

```diff
@@ -48,49 +52,54 @@ class ValidatorStakeProcessor @Inject constructor(
     private suspend fun DetailedManifestClass.ValidatorStake.extractDeposits(
         executionSummary: ExecutionSummary,
         getProfileUseCase: GetProfileUseCase,
-        resources: List<Resource>,
+        assets: List<Asset>,
         involvedValidators: List<ValidatorDetail>
     ) = executionSummary.accountDeposits.map { depositsPerAccount ->
         val ownedAccount = getProfileUseCase.accountOnCurrentNetwork(depositsPerAccount.key) ?: error("No account found")
         val defaultDepositGuarantees = getProfileUseCase.invoke().first().appPreferences.transaction.defaultDepositGuarantee
-        val depositingLsu = depositsPerAccount.value.groupBy { it.resourceAddress }.map { depositedResources ->
-            val lsuResource = resources.find {
-                it.resourceAddress == depositedResources.key
-            } as? Resource.FungibleResource ?: error("No resource found")
-            val relatedStakes = validatorStakes.filter { it.liquidStakeUnitAddress.addressString() == lsuResource.resourceAddress }
-            val totalStakedLsuForAccount = relatedStakes.sumOf { it.liquidStakeUnitAmount.asStr().toBigDecimal() }
-            val totalStakeXrdWorthForAccount = relatedStakes.sumOf { it.xrdAmount.asStr().toBigDecimal() }
-            val validatorAddress = lsuResource.validatorAddress ?: error("No validator address found")
-            val validator =
-                involvedValidators.find { it.address == validatorAddress } ?: error("No validator found")
-            val lsuAmount = depositedResources.value.sumOf { it.amount }
-            val xrdWorth = lsuAmount.divide(totalStakedLsuForAccount, lsuResource.mathContext)
-                .multiply(totalStakeXrdWorthForAccount, lsuResource.mathContext)
-            val guaranteeType = depositedResources.value.first().guaranteeType(defaultDepositGuarantees)
-            Transferable.Depositing(
-                transferable = TransferableAsset.Fungible.LSUAsset(
-                    amount = lsuAmount,
-                    lsu = LiquidStakeUnit(lsuResource.copy(ownedAmount = lsuAmount), validator),
-                    xrdWorth = xrdWorth,
-                ),
-                guaranteeType = guaranteeType
-            )
+        val depositingTransferable = depositsPerAccount.value.map { depositedResource ->
+            val resource = assets.find {
+                it.resource.resourceAddress == depositedResource.resourceAddress
+            }?.resource ?: error("No resource found")
+            val validatorAddress = resource.validatorAddress
+            if (validatorAddress == null) {
+                executionSummary.resolveGeneralAsset(depositedResource, assets, defaultDepositGuarantees)
```

Comment:
I do not have too much knowledge on these transactions, but as far as I understand there are a couple of points here. Correct me if I am wrong.
- `ValidatorStakeProcessor` and `PoolContributionProcessor` can only deal with `Fungible` and not with `NonFungible` . At least this is what these two processors return here. `resource as? Resource.FungibleResource ?: error("No fungible resou...`
- based on the above the extension fun `resolveGeneralAsset` should be more type safe and accept only `resourceIndicator: ResourceIndicator.Fungible`

Reply:
this is update, assumption is no longer there

Reply:
sorry I am not sure what you mean?

Reply:
in `ValidatorStakeProcessor` `resolveGeneralAsset` call no longer takes Resource.NonFungible as input parameter, but `Resource`.

Reply:
Right, and that was my question. Why it takes `Resource` and not `Resource.Fungible`? Can you stake or have pool units with NFTs? As far as I know no. 
Then based on this assumption that's why I asked why we need to pass `Resource`? Wouldn't be safer to have it as `Resource.Fungible`?

Reply:
apparently if transaction is a stake/pool related, it can still have resources that are not pools and LSUs - for example you have stake tx but also in same manifest there is "regular" XRD or NFT transfer along staking. So it should be Resource here, and we should also update all the processors to not assume that stake/pool tx does not have any other resources.